Staging eli (shapefile regions)

zebbecker opened the pull request:
Adds ability to filter inputs based on custom region shapefiles rather than bounding boxes.

eorland committed:
changes for reading in global shapefiles (wip)

eorland committed:
update AFP_regfilter

eorland committed:
make AFP_regfilter return in lat lon

eorland committed:
change filepath

eorland committed:
change shapefile folder name

zebbecker committed:
Merge branch 'staging' into staging-eli

Grab latest updates from conus-dps.

eorland committed:
update consts

eorland committed:
update fire consts

zebbecker committed:
Prep for merge to staging

zebbecker committed:
Restore string option for CRS env var

zebbecker commented:
@eorland we have a little more work to do before merging this to staging and I figured I'd open this PR to give us a space to coordinate. 

The changes here cause some of our tests to fail, which is not necessarily unexpected since we are legitimately changing functionality, but which we should nonetheless address.

 
<img width="774" height="208" alt="Screenshot 2025-07-15 at 11 17 34 AM" src="https://github.com/user-attachments/assets/1c8e09fc-21f5-4b56-8d8e-4790d5c8281d" />

For easy reference, here is how to run the tests, assuming you have this branch checked out: 

```
# create and activate virtual environment 
python -m venv env 
source env/bin/activate

# install optional test dependencies in addition to the regular ones
python -m pip install -e '.[test]'

# run tests 
python -m pytest 
```

Could you please take a stab at fixing the tests so that they don't try to call the dead code you (rightly) removed, but do test the new functionality you added? Happy to collab on that as needed, just Slack me.

zebbecker committed:
Add docstring for CLI script

zebbecker commented on fireatlas/FireIO.py:
```diff
@@ -801,16 +803,9 @@ def AFP_regfilter(df, shp_Reg):
     df_filtered : pandas DataFrame
         the filtered fire pixels
     """
-    # preliminary spatial filter and quality filter
```
@eorland I'd suggest keeping this preliminary filter in, as it is much faster to do the basic bbox filtering on the vanilla pandas dataframe without any geographic operations. That way, we avoid projecting and checking geographically (expensive!) pixels from totally different continents, for example. I haven't tested it with the new shapefile setup, though, so if I'm missing another reason to make this change please do let me know.

zebbecker commented on fireatlas/FireIO.py:
```diff
@@ -801,16 +803,9 @@ def AFP_regfilter(df, shp_Reg):
     df_filtered : pandas DataFrame
         the filtered fire pixels
     """
-    # preliminary spatial filter and quality filter
-    regext = shp_Reg.bounds
-    newfirepixels = df.loc[
-        (df["Lat"] >= regext[1])
-        & (df["Lat"] <= regext[3])
-        & (df["Lon"] >= regext[0])
-        & (df["Lon"] <= regext[2])
-    ]
```
@eorland I'd suggest keeping this preliminary filter in, as it is much faster to do the basic bbox filtering on the vanilla pandas dataframe without any geographic operations. That way, we avoid projecting and checking geographically (expensive!) pixels from totally different continents, for example. I haven't tested it with the new shapefile setup, though, so if I'm missing another reason to make this change please do let me know.

eorland committed:
add initial lat lon filtering in afp_regfilter

eorland commented on fireatlas/FireIO.py:
```diff
@@ -801,16 +803,16 @@ def AFP_regfilter(df, shp_Reg):
-    regext = shp_Reg.bounds
-    newfirepixels = df.loc[
+
+    regext = shp_Reg.total_bounds
```
@zebbecker added this back in!

eorland commented on fireatlas/FireIO.py:
```diff
@@ -1028,10 +1014,11 @@ def get_reg_shp(reg):
     if isinstance(reg, shapely.geometry.base.BaseGeometry):
         shp_Reg = reg
```
@zebbecker something else I noticed (but doesn't break anything yet): this assumes the reg variable is in EPSG:4326. I am not aware of any workflow which meets this condition but it could hypothetically cause issues down the line. I'm wondering if we either remove it, or add in a way to make sure the crs is correct?

Currently, we only input file name strings (condition 2) or the bbox lists (condition 3), so I wonder if we eventually try to be more specific about type checking in this function?

zebbecker replied:
Good catch. Agree that this is prime bug surface area. I looked into just removing it now, but the Region code is messy and spread across the codebase. I opened #212 to remind myself to deal with this when time allows

eorland committed:
fix bounds error

eorland committed:
edit reg tests (WIP)

zebbecker committed:
Test reading geometry from file

zebbecker commented:
@eorland that last failing test was expecting `get_reg_shp` to return None if it was provided an invalid file name, but what it actually does (and, I think, what it should do) is throw an error if the shapefile is not found. I changed that test, and then also added a test to make sure that if you pass it a _valid_ filename, it does return the geometry from that file.

zebbecker commented:
Lgtm! Thanks for pushing this through @eorland
